.github/workflows/do-not-merge-checker.yml

```diff
@@ -0,0 +1,22 @@
+# This workflow fails if a 'do-not-merge' label is applied to the PR.
+name: Check do-not-merge
+
+on:
+  pull_request:
+    types: [reopened, labeled, unlabeled]
+    # Runs on PRs to main and release branches
+    branches:
+      - main
+      - release/**
+
+jobs:
+  # checks that a do-not-merge label is not present for a PR
+  do-not-merge-check:
+    # If there is a `do-not-merge` label, or this comes from a fork (community contributor) we ignore this check
+    if: ${{ (github.repository_owner == 'hashicorp' && contains(github.event.pull_request.labels.*.name, 'do-not-merge')) }}
```

Comment:
Have we 100% validated that this works? In the past, new GHAs failing on community branches has been a persistent issue, so I want to make sure that we're not introducing a new issue here

Reply:
Oh, that's a fair point. In the past, we've stumbled for stuff like community submitted PRs don't have the permissions to run a GHA that adds a label. If community PRs have permissions to read labels, then I guess I don't see why not, yeah. There's some stuff that shouldn't run, like the milestone check (which essentially never applies to community PRs), but if this one runs without failing, I see no harm in it!

Reply:
Thank you!